src/rust/ide/view/graph-editor/src/component/visualization/container.rs

```diff
@@ -602,8 +602,12 @@ impl Container {
             eval_ on_selected ( action_bar.hide_icons.emit(()) );
         }
 
+        // FIXME[mm]: If we set the size right here, we will see spurious shapes in some
+        // computation heavy circumstances (e.g., collapsing nodes, or creating an new project).
+        // If we let the animation do the size change, the size will only be set after the delays.
+        // This is not optimal the optimal solution to this problem, as it also means that we have
+        // an animation on an invisible component running.
```

Comment:
add issue link to describe it here better. Moreover, I do not fully understand the reasons for it. Could you write this in more strucured way - like in points 1. ..., 2 ... , 3... - what happens after each other and why?

Reply:
I've expanded the description.